src/service/alerts/scheduler.rs

```diff
@@ -773,70 +778,78 @@ async fn handle_derived_stream_triggers(
     );
     let (_, max_retries) = get_scheduler_max_retries();
 
-    // module_key format: stream_type/stream_name/pipeline_name/derived_stream_name
+    // module_key format: stream_type/org_id/pipeline_name/pipeline_id
     let columns = trigger.module_key.split('/').collect::<Vec<_>>();
     assert_eq!(columns.len(), 4);
-    let org_id = &trigger.org;
     let stream_type: StreamType = columns[0].into();
-    let stream_name = columns[1];
+    let org_id = columns[1];
     let pipeline_name = columns[2];
-    let name = columns[3];
+    let pipeline_id = columns[3];
 
     let is_real_time = trigger.is_realtime;
     let is_silenced = trigger.is_silenced;
+
+    let mut new_trigger = db::scheduler::Trigger {
+        next_run_at: Utc::now().timestamp_micros(),
+        is_silenced: false,
+        status: db::scheduler::TriggerStatus::Waiting,
+        retries: 0,
+        ..trigger.clone()
+    };
+
     if is_real_time && is_silenced {
         log::debug!(
             "Realtime derived_stream needs to wake up, {}/{}",
             org_id,
             trigger.module_key
         );
-        let new_trigger = db::scheduler::Trigger {
-            next_run_at: Utc::now().timestamp_micros(),
-            is_silenced: false,
-            status: db::scheduler::TriggerStatus::Waiting,
-            ..trigger.clone()
-        };
         db::scheduler::update_trigger(new_trigger).await?;
         return Ok(());
     }
 
-    let Ok(pipeline) = db::pipelines::get(org_id, stream_type, stream_name, pipeline_name).await
-    else {
+    let Ok(pipeline) = db::pipeline::get_by_id(pipeline_id).await else {
+        log::warn!(
+            "Pipeline associated with trigger not found: {}/{}/{}/{}. Deleting this trigger",
+            org_id,
+            stream_type,
+            pipeline_name,
+            pipeline_id
+        );
+        db::scheduler::delete(&trigger.org, trigger.module, &trigger.module_key).await?;
         return Err(anyhow::anyhow!(
             "Pipeline associated with trigger not found: {}/{}/{}/{}",
             org_id,
-            stream_name,
             stream_type,
-            pipeline_name
+            pipeline_name,
+            pipeline_id
         ));
     };
 
-    let Some(derived_stream) = pipeline
-        .derived_streams
-        .and_then(|ds| ds.into_iter().find(|ds| ds.name == name))
-    else {
+    if !pipeline.enabled {
+        log::info!("Pipeline associated with trigger not enabled.");
+        db::scheduler::update_trigger(new_trigger).await?;
```

Comment:
@taimingl I think here the `new_trigger.next_run_at` needs to be updated to a value (lets say 7 days later). Otherwise, since the `next_run_at` of `new_trigger` is `Utc::now().timestamp_micros()`, scheduler will pull the trigger again and again.